packages/walletconnect/src/index.ts

```diff
@@ -73,81 +55,81 @@ export class WalletConnect extends Connector {
     if (error) this.onError?.(error)
   }
 
-  private chainChangedListener = (chainId: number | string): void => {
-    this.actions.update({ chainId: parseChainId(chainId) })
+  private chainChangedListener = (chainId: number): void => {
+    this.actions.update({ chainId })
   }
 
   private accountsChangedListener = (accounts: string[]): void => {
     this.actions.update({ accounts })
   }
 
-  private URIListener = (_: Error | null, payload: { params: string[] }): void => {
-    this.events.emit(URI_AVAILABLE, payload.params[0])
+  private URIListener = (uri: string): void => {
+    this.events.emit(URI_AVAILABLE, uri)
   }
 
-  private async isomorphicInitialize(chainId = this.defaultChainId): Promise<void> {
-    if (this.eagerConnection) return
+  private async isomorphicInitialize(): Promise<MockWalletConnectProvider> {
+    if (this.eagerConnection) return this.eagerConnection
 
-    // because we can only use 1 url per chainId, we need to decide between multiple, where necessary
-    const rpc = Promise.all(
-      Object.keys(this.rpc).map(
-        async (chainId): Promise<[number, string]> => [
-          Number(chainId),
-          await getBestUrl(this.rpc[Number(chainId)], this.timeout),
-        ]
+    if (this.provider) {
+      await this.deactivate()
+    }
+    
+    const rpcMap = this.rpcMap
+    const resolvedRpcMap = rpcMap ? Object.fromEntries(
```

Comment:
would `rpcBestUrlMap` be more clear? this is a bit of a nest, and my read of it is that for each rpc in the map you try to find the best url using this function. with repeated names i don't like to change types and it appears that the values here are individual strings vs arrays of strings (unless i'm reading this wrong)

Reply:
Sounds good! This is exactly what this function does, but I am yet to describe it at the beginning of the file (where I left a todo to describe reason for `rpcMap`).